src/psp.ts

```diff
@@ -0,0 +1,502 @@
+
+        // **************************************************************************
+        // Fetch subscriptions from allocators
+        // **************************************************************************
+        node.logger.debug('Fetching subcriptions...', providerLogOptions);
+
+        //TODO: where to get abi from?
+        const allocatorAbi = [
+          {
+            inputs: [
+              {
+                internalType: 'address',
+                name: '',
+                type: 'address',
+              },
+              {
+                internalType: 'uint256',
+                name: '',
+                type: 'uint256',
+              },
+            ],
+            name: 'airnodeToSlotIndexToSlot',
+            outputs: [
+              {
+                internalType: 'bytes32',
+                name: 'subscriptionId',
+                type: 'bytes32',
+              },
+              {
+                internalType: 'address',
+                name: 'setter',
+                type: 'address',
+              },
+              {
+                internalType: 'uint64',
+                name: 'expirationTimestamp',
+                type: 'uint64',
+              },
+            ],
+            stateMutability: 'view',
+            type: 'function',
+          },
+          {
+            inputs: [
+              {
+                internalType: 'bytes[]',
+                name: 'data',
+                type: 'bytes[]',
+              },
+            ],
+            name: 'multicall',
+            outputs: [
+              {
+                internalType: 'bytes[]',
+                name: 'results',
+                type: 'bytes[]',
+              },
+            ],
+            stateMutability: 'nonpayable',
+            type: 'function',
+          },
+        ];
+
+        const subscriptionIds: string[] = [];
+        for (const { address, startIndex, endIndex } of chain.allocators) {
+          const allocator = new ethers.Contract(address, allocatorAbi, provider);
+          const staticCalldatas = [];
+          for (let i = startIndex; i <= endIndex; i++) {
+            staticCalldatas.push(
+              allocator.interface.encodeFunctionData('airnodeToSlotIndexToSlot', [airnodeAddress, i])
+            );
+          }
+          // TODO: retryGo?
+          const resultDatas = await allocator.callStatic.multicall(staticCalldatas);
+          resultDatas.forEach((resultData: ethers.utils.BytesLike) => {
+            const result = allocator.interface.decodeFunctionResult('airnodeToSlotIndexToSlot', resultData);
+            if (result.subscriptionId && result.expirationTimestamp.gt(nowInSeconds())) {
+              subscriptionIds.push(result.subscriptionId);
+            }
+          });
+        }
+
+        // **************************************************************************
+        // Process each psp subscription in serial to keep nonces in order
```

Comment:
You should only process subscriptions serially (sorted by ID) if they use the same sponsor

Reply:
Is there really a value to sort them by subscriptionId?